Shrinkray/46/fix semantic html

shrinkray opened the pull request:
# Pull Request

## Description

Comprehensive accessibility and semantic HTML improvements across the DevFest website. This PR addresses multiple WCAG compliance issues and enhances the user experience for all users, including those using assistive technologies.

## Type of Change

<!-- Mark the relevant option with an "x" -->

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Code refactoring
- [ ] Performance improvement
- [x] Accessibility improvement
- [ ] Other (please describe):

## Related Issues

Fixes #46  semantic HTML issues identified during accessibility audit
Addresses WCAG 2.1 AA compliance gaps
Improves navigation and landmark structure

## Code Quality Checklist

<!-- Mark completed items with an "x" -->

- [x] Code follows the project's coding standards
- [x] Self-review of the code has been performed
- [x] Code is properly commented where necessary
- [x] No console.log statements left in the code
- [x] No hardcoded values that should be configurable
- [x] Error handling is implemented where appropriate

## Testing

<!-- Mark completed items with an "x" -->

- [x] Changes have been tested locally
- [x] All existing tests pass (if applicable)
- [ ] New tests have been added for new functionality (if applicable)
- [x] Manual testing has been performed
- [x] Accessibility testing has been performed (for UI changes)

## Accessibility (for UI changes)

<!-- Mark completed items with an "x" -->

- [x] All interactive elements are keyboard accessible
- [x] Color contrast meets WCAG AA standards
- [x] Images have appropriate alt text
- [x] Form labels are properly associated
- [x] Focus indicators are visible
- [x] Screen reader compatibility has been tested

## Build & Deployment

<!-- Mark completed items with an "x" -->

- [x] Application builds successfully
- [x] No build warnings or errors
- [x] Dependencies are up to date
- [x] No security vulnerabilities introduced

## Screenshots (if applicable)

### Locations button for directions
<img width="1740" height="560" alt="Screenshot 2025-09-29 at 11 43 10 AM" src="https://github.com/user-attachments/assets/8edb687a-ee2d-4cea-8bbe-4bd3a712942f" />

### Centered headings
<img width="1723" height="571" alt="Screenshot 2025-09-29 at 11 51 34 AM" src="https://github.com/user-attachments/assets/e5fe7556-aabd-479d-82f6-634762378c60" />

## Additional Notes

Made a few changes that address aesthetics. 
Work on navbar was addressed on the mobile-navbar-fix branch

---

**Note**: This PR will be automatically checked for:

- Code quality (ESLint)
- Code formatting (Prettier)
- Build verification
- Accessibility compliance
- Security vulnerabilities

All checks must pass before this PR can be merged.

shrinkray committed:
refactor: use semantic heading elements

- replace <header></header> with <h2></h2>

shrinkray committed:
refactor: center headings

- position icons with absolute
- headings look on center
- scales and positioned for mobile

shrinkray committed:
refactor: fix landmarks

- improve heading hierarchy across all pages
- clear landmarks
- screen reader nav
- SEO improvement
- WCAG compliance guidelines

shrinkray committed:
refactor: fix card hierarchy elements

- h1 improperly used
- fix heading cascade

shrinkray committed:
refactor: help for screenreader nav

- nav by headings and landmarks
- clear labels for buttons and controls
- modal accessibility, use dialog semantics
- expandable content
- WCAG compliance

shrinkray committed:
refactor: make address accessible

- use address element
- add get directions button for cleaner hit target

shrinkray committed:
refactor: increase contrast

- move button text color to text-primary-50 for stronger contrast
- Contrast is 8:1+ exceeds WCAG AAA standards
- brand consistency
-

shrinkray committed:
fix: set text color for contrast

shrinkray committed:
refactor: add padding to bottom of row

- give button room

ShugKnight24 commented on src/pages/Home.jsx:
```diff
@@ -11,7 +11,10 @@ import SponsorsSection from '@/layouts/SponsorsSection'
 function Home() {
   return (
-    <main>
+    <main aria-labelledby="main-heading">
```
This element was removed [here](https://github.com/GDG-Detroit/devfest-website/pull/45/files)

The accompanying change was made in PageLayout, which this will now use

ShugKnight24 reviewed (commented):
This is a good PR - I think we've deviated from 2025 a too much in the PRs we have up currently. We should get things in sync before making these changes creating conflicts and potentially overwriting previous work depending on merge order

shrinkray committed:
Merge branch '2025' into shrinkray/46/fixSemanticHtml

shrinkray committed:
refactor: update NotFound.jsx

- remove <main> boilerplate
- add PageLayout
- now consistent with other pages

shrinkray commented:
Should be up to date with branch.

ShugKnight24 commented on src/components/speakers/SpeakerCard.jsx:
```diff
@@ -30,7 +30,13 @@ const SpeakerCard = ({
       />
 
       {isModalOpen && id === speakerID && (
-        <div className="fixed inset-0 z-50 flex items-center justify-center bg-gray-800/75">
+        <div
+          className="fixed inset-0 z-50 flex items-center justify-center bg-gray-800/75"
+          role="dialog"
+          aria-modal="true"
+          aria-labelledby="speaker-modal-title"
+          aria-describedby="speaker-modal-bio"
+        >
```
I changed the structure of this in #42 so there might be conflicts that arise depending on which gets merged first

ShugKnight24 commented on src/pages/NotFound.jsx:
```diff
@@ -11,18 +12,20 @@ const NotFound = () => {
-        Return to Home Page
-      </NavLink>
-    </div>
+    <PageLayout>
```
I think wrapping this in page layout is good and keeps things consistent

If we're adding now a Navbar here, we should probably do some conditional rendering like @JustinBeaudry is doing in #30 as most of the nav is tailored for the home page

A bit of a moot point as the user will get redirected there

ShugKnight24 reviewed (commented):
Good PR, addresses accessibility warnings we're currently seeing

Might have some conflicts depending on merge order & think we should address the Navbar in the future

I'll let @JustinBeaudry run his cursor bot etc

shrinkray committed:
Merge branch '2025' into shrinkray/46/fixSemanticHtml

Resolved merge conflicts by choosing 2025 version for:
- SpeakerCard.jsx: Better modal structure with backdrop blur
- SpeakerDetails.jsx: Improved UI with better layout and navigation
- GenericCard.jsx: Enhanced styling with hover effects
- SpeakersSection.jsx: Uses new SpeakersContent component
- SponsorsSection.jsx: Added year support for historical events
- PreviousEvents.jsx: Uses new EventList component

All conflicts resolved successfully.

shrinkray committed:
refactor: navigation

- Add role menubar and role menuitem to desktop nav
- Add aria-label for Main navigation
- Add screen reader announcements with aria-live polite
- Add aria-current page for active items
- Add aria-expanded, aria-controls and aria-labelledby for mobile nav
- Add role none for li elements
- Standardized PreviousEventsNavBar to use same styles as others
- Replace yellow highlight with thick border animation
- Add padding to all nav elements
- Increase breakpoint trigger to capture nav earlier
- Fix tailwind class order
- Resolve lint bugs
Consistent structure
Improve semantic html

shrinkray committed:
refactor: speakercard

- Fix focus management
- ARIA attributes
  - role dialog
  - aria-modal true
  - aria-labelledby and aria-describedby
  - aria-label on on buttons
- Keyboard navigatiion
- Screen reader support
  - unique ids
  - Descriptive labels for interactive elements
  - heading hierarchy
- Non-interactive elements
  - move event handlers to dedicated interactive element
  - button is interactive will receive mouse and keyboard events

shrinkray committed:
refactor: headings

- semantic headings
- h2 centering and icon position
- unify padding at each section

shrinkray committed:
refactor: modal

- h2 has modal title
- h3 for subsections
- screen reader to navigate correctly
- ARIA aria-labelledby points to the h2 modal title

shrinkray committed:
refactor: location

- Headings were out of order
- Add semantic details for date and location info

shrinkray committed:
refactor: color contrast

- increase contrast on link element

shrinkray committed:
refactor: previous event page

- Add hover treatment to link
- add aria-label

shrinkray committed:
refactor: landmarks

ARIA landmarks added
- role navigation to previousEvents
- EventList banner, region, main-heading, not found heading
- benefits screen readers, structure

shrinkray committed:
refactor: landmarks

- banner landmark should not be contained in another landmark
- all pages contained by landmarks
- skip link must be contained by landmark
  - allow app to contain previous event and EventList
- fix skip-link main target
- add #main-content to to the main element
- pages affected hoome, previous events individual event and 404 via pageLayout

shrinkray commented:
@ShugKnight24 Complete review and fixed all areas, all pass

JustinBeaudry reviewed (approved):
Great work @shrinkray!
